fw/cache.c

```diff
@@ -2265,10 +2285,18 @@ tfw_cache_do_action(TfwHttpMsg *msg, tfw_http_cache_cb_t action)
 	}
 
 	req = (TfwHttpReq *)msg;
-	if (unlikely(req->method == TFW_HTTP_METH_PURGE))
-		tfw_cache_purge_method(req);
-	else
+	if (unlikely(req->method == TFW_HTTP_METH_PURGE)) {
+		if (tfw_cache_purge_method(req))
+			return;
+		if (unlikely(req->cache_ctl.flags & TFW_HTTP_CC_CACHE_PURGE)) {
+			cache_req_process_node(req, action);
+		} else {
+			tfw_http_send_resp(req, 200, "purge: success");
+		}
+	}
+	else {
 		cache_req_process_node(req, action);
+	}
```

Comment:
This is essencially just
```C
        if (unlikely(req->method == TFW_HTTP_METH_PURGE)) {
                if (tfw_cache_purge_method(req))
                        return;
                if (!(req->cache_ctl.flags & TFW_HTTP_CC_CACHE_PURGE)) {
                        tfw_http_send_resp(req, 200, "purge: success");
                        return;
                }
        }
        cache_req_process_node(req, action);
```
which is simpler. We don't need likely/unlikely here because we're already in `unlikely` branch

Reply:
I took this variant. done.